spring-integration-core/src/main/java/org/springframework/integration/support/management/MessageChannelMetrics.java

```diff
@@ -20,14 +20,18 @@
 import org.springframework.jmx.support.MetricType;
 
 /**
- * Interface for all message channel monitors containing accessors for various useful metrics that are generic for all
- * channel types.
+ * Interface for all message channel monitors containing accessors for various useful
+ * metrics that are generic for all channel types.
  *
  * @author Dave Syer
  * @author Gary Russell
+ * @deprecated in favor of dimensional metrics via
```

Comment:
You know I have a thought do not deprecate `*Metrics` classes, but make them to delegate to the meters from the Micrometer. I think it is still good to have actual metrics exposed from the components per se.
See `HistogramSupport.takeSnapshot()`. Looks like those `count`, `total`, `max` are usable for us.

I haven't figure out how to properly deprecate the `MetricsFactory` not removing too much test classes.
And also I didn't understand if that make sense to deprecate `ExponentialMovingAverage*` family...

I'll take a look tomorrow closer what you have so far.

Thanks for the effort!

Reply:
Yeah, that's why I added the new super interfaces, so we can still expose counts via the meters. If there are other interesting metrics we can add, we can put them there; either in a future 5.2 milestone, or when we pull the legacy metrics.